Modules/timemodule.c

```diff
@@ -1759,52 +1759,78 @@ PyInit_time(void)
 
     /* Set, or reset, module variables like time.timezone */
     if (init_timezone(m) < 0) {
-        return NULL;
+        goto error;
     }
 
 #if defined(HAVE_CLOCK_GETTIME) || defined(HAVE_CLOCK_SETTIME) || defined(HAVE_CLOCK_GETRES)
 
 #ifdef CLOCK_REALTIME
-    PyModule_AddIntMacro(m, CLOCK_REALTIME);
+    if (PyModule_AddIntMacro(m, CLOCK_REALTIME) < 0) {
```

Comment:
Is the reason that you need `goto` in every one of these blocks that `PyErr_Occcurred()` will get reset if you make other `PyModule_AddIntMacro` calls that don't fail along the way?

Either way, I'm mildly tempted to suggest adding a wrapper macro like:

```c
#define _PyModule_AddIntMacroOrGotoError(m, macro) \
    if (PyModule_AddIntMacro(m, macro) < 0) { goto error; }
```

But I dunno. I'm torn between the principles of "don't repeat yourself" and "don't write a preprocessor macro with a goto in it (you monster)".

Reply:
Between the two options, I'm personally +0 on repetition. Incorrect error handling for the whole `PyModule_Add` family of functions is rampant in the stdlib (see my work in bpo-38823), and I think reinforcing the correct usage is good to have to avoid people repeating the same old patterns (especially in outside projects).

Reply:
>Is the reason that you need goto in every one of these blocks that PyErr_Occcurred() will get reset if you make other PyModule_AddIntMacro calls that don't fail along the way?

No, I just try to make the `Py_DECREF(m);return NULL;` looks like a common processing item.

>I think reinforcing the correct usage is good to have to keep people from repeating the same old patterns

+1

Reply:
I dislike macros hiding a goto. A goto statement is dangerous, I prefer to make it explicit. The "if (...) { goto error; }" is a very common pattern in Python, especially in initialization code.